src/pages/iou/steps/MoneyRequestConfirmPage.js

```diff
@@ -61,30 +63,37 @@ const defaultProps = {
 };
 
 function MoneyRequestConfirmPage(props) {
-    const prevMoneyRequestId = useRef(props.iou.id);
+    const prevMoneyRequestId = usePrevious(props.iou.id);
     const iouType = useRef(lodashGet(props.route, 'params.iouType', ''));
     const reportID = useRef(lodashGet(props.route, 'params.reportID', ''));
+    const participants = useMemo(
+        () =>
+            lodashGet(props.iou, ['participants', 0, 'isPolicyExpenseChat'], false)
+                ? OptionsListUtils.getPolicyExpenseReportOptions(props.iou.participants[0])
+                : OptionsListUtils.getParticipantsOptions(props.iou.participants, props.personalDetails),
+        // The rule can't recognize that we are accessing participants from props.iou, so let's ignore the warning
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [props.iou.participants, props.personalDetails],
+    );
 
     useEffect(() => {
-        const moneyRequestId = `${iouType.current}${reportID.current}`;
         // ID in Onyx could change by initiating a new request in a separate browser tab
-        const isMoneyRequestIdChange = prevMoneyRequestId.current !== props.iou.id;
-        const isMoneyRequestIdMatch = props.iou.id === moneyRequestId;
+        if (prevMoneyRequestId && prevMoneyRequestId !== props.iou.id) {
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
+            return;
+        }
 
         // Reset the money request Onyx if the ID in Onyx does not match the ID from params
-        // and is not caused by an ID change in Onyx.
-        if (!isMoneyRequestIdMatch && !isMoneyRequestIdChange) {
+        const moneyRequestId = `${iouType.current}${reportID.current}`;
+        const shouldReset = props.iou.id !== moneyRequestId;
+        if (shouldReset) {
             IOU.resetMoneyRequestInfo(moneyRequestId);
         }
 
-        if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || !isMoneyRequestIdMatch || isMoneyRequestIdChange) {
+        if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || shouldReset) {
             Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
```

Comment:
```suggestion
            Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
            return;
```

have no impact but it's a good pattern to always return on changing navigation in case someone wrote code below this block.

Reply:
I agree, but unfortunately the lint doesn't agree 😢 
```
error  Unnecessary return statement  no-useless-return
```

Reply:
oh okay, nothing to do then